lightningd/pay.c

```diff
@@ -946,6 +969,20 @@ static struct command_result *json_sendonion(struct command *cmd,
 				    "with failcode=%d",
 				    failcode);
 
+	if (secretstok) {
+		path_secrets = tal_arr(cmd, struct secret, secretstok->size);
+		json_for_each_arr(i, cur, secretstok) {
+			if (!json_to_secret(buffer, cur, &path_secrets[i]))
+				return command_fail(
+				    cmd, JSONRPC2_INVALID_PARAMS,
+				    "shared_secret[%zu] isn't a valid "
+				    "hex-encoded 32 byte secret",
+				    i);
+		}
+	} else {
+		path_secrets = NULL;
+	}
```

Comment:
Again, it's be nice to implement this as param_secrets_array so 'check' rpc works better.